core/src/main/java/hudson/slaves/Cloud.java

```diff
@@ -310,8 +317,68 @@ public HttpResponse doDoDelete() throws IOException {
         return new HttpRedirect("..");
     }
 
+
+    @Override
+    public boolean isNameEditable() {
+        return true;
+    }
+
+    @RequirePOST
+    @Restricted(NoExternalUse.class)
+    @Override
+    public HttpResponse doConfirmRename(@QueryParameter String newName) throws IOException, ServletException, Descriptor.FormException {
+        newName = newName == null ? null : newName.trim();
+        FormValidation validationError = doCheckNewName(newName);
+        if (validationError.kind != FormValidation.Kind.OK) {
+            throw new Failure(validationError.getMessage());
+        }
+        this.name = newName;
```

Comment:
I suspect this is an issue. I think I need to recreate a new instance of the object in order to have the inherited name (like `AzureVMCloud.cloudName`) correctly applied